packages/theme/src/cli/services/dev.ts

```diff
@@ -150,6 +154,12 @@ export async function dev(options: DevOptions) {
         openURLSafely(urls.giftCard, 'gift card preview')
         break
     }
+
+    // This prevents the keypress handler from being called multiple times in a short span
+    keyBlock = true
+    setTimeout(() => {
```

Comment:
Are you familiar with [debouncing](https://developer.mozilla.org/en-US/docs/Glossary/Debounce)? We have a `debounce` helper function that we could potentially use here instead? Might be overkill for what you're doing but it's there if you want! https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/public/common/function.ts#L39

We  could use `leading: true` to make sure we don't delay the first keypress.